psa-crypto/src/operations/key_management.rs

```diff
@@ -307,3 +307,99 @@ fn complete_new_key_operation(
         },
     })
 }
+
+/// Copy key material from one location to another
+/// The function returns the key ID of the newly created key `Id` can be set to `None` when creating a volatile key.
+/// When generating a persistent key with a specific ID, the `Id` structure can be created after
+/// reset with the `from_persistent_key_id` constructor on `Id`.
+///
+/// The originating key must have the usage flag `copy` set.
+///
+/// # Example
+///
+/// ```
+/// # use psa_crypto::operations::key_management;
+/// # use psa_crypto::types::key::{Attributes, Type, Lifetime, Policy, UsageFlags};
+/// # use psa_crypto::types::algorithm::{AsymmetricSignature, Hash};
+/// # let mut attributes = Attributes {
+/// #     key_type: Type::RsaKeyPair,
+/// #     bits: 1024,
+/// #     lifetime: Lifetime::Volatile,
+/// #     policy: Policy {
+/// #         usage_flags: UsageFlags {
+/// #             copy: true,
+/// #             ..Default::default()
+/// #         },
+/// #         permitted_algorithms: AsymmetricSignature::RsaPkcs1v15Sign {
+/// #             hash_alg: Hash::Sha256.into(),
+/// #         }.into(),
+/// #     },
+/// # };
+/// psa_crypto::init().unwrap();
+/// let my_key = key_management::generate(attributes, None).unwrap();
+/// let my_key_copy = key_management::copy_key(my_key, attributes, None);
+/// ```
+pub fn copy_key(key_id_to_copy: Id, attributes: Attributes, id: Option<u32>) -> Result<Id> {
```

Comment:
When thinking about it, we removed the "key" from all function names: this should probably be `copy`. Similar for `export_key` above, it should maybe be `export`. And `purge` below.